core/modules/replica/HttpIngestChunksModule.cc

```diff
@@ -467,4 +468,62 @@ void HttpIngestChunksModule::_registerNewChunk(string const& worker,
     controller()->serviceProvider()->databaseServices()->saveReplicaInfo(newReplica);
 }
 
+
+json HttpIngestChunksModule::_getChunks() {
+    debug(__func__);
+
+    auto const databaseServices = controller()->serviceProvider()->databaseServices();
+    auto const config = controller()->serviceProvider()->config();
+
+    string const database = body().required<string>("database");
+    debug(__func__, "database=" + database);
+
+    auto const databaseInfo = config->databaseInfo(database);
+
+    // Locate replicas (if any) for all chunks of the database.
+    bool const enabledWorkersOnly = true;
+    vector<ReplicaInfo> replicas;
+    databaseServices->findDatabaseReplicas(replicas, database, enabledWorkersOnly);
+
+    // Build the chunk-to-worker map to be returned to a client in the result
+    // object. Note that published databases may have more than 1 replica of a chunk.
+
+    json result;
+    result["replica"] = json::array();
+
+    for (auto const& replica: replicas) {
+
+        json replicaResult;
+        replicaResult["chunk"] = replica.chunk();
+        replicaResult["worker"] = replica.worker(); 
+
+        // Initialize required attributes to the default values (all 0) for all
+        // relevant tables.
+        for (auto&& table: databaseInfo.partitionedTables) {
+            replicaResult[table]["overlap_rows"] = 0;          // TBC when available
+            replicaResult[table]["overlap_data_size"] = 0;
+            replicaResult[table]["overlap_index_size"] = 0;
+
+            replicaResult[table]["rows"] = 0;                  // TBC when available
+            replicaResult[table]["data_size"] = 0;
+            replicaResult[table]["index_size"] = 0;
+        }
+
+        // fetch and report actual values of the attributes.
+        for (auto&& file: replica.fileInfo()) {
+            auto& replicaTableResult = replicaResult[file.baseTable()];
+            if (file.isOverlap()) {
+                if      (file.isData())  replicaTableResult["overlap_data_size"]  = file.size;
+                else if (file.isIndex()) replicaTableResult["overlap_index_size"] = file.size;
+            } else {
+                if      (file.isData())  replicaTableResult["data_size"]  = file.size;
+                else if (file.isIndex()) replicaTableResult["index_size"] = file.size;
```

Comment:
Multi-line `if` should use curly braces: https://developer.lsst.io/cpp/style.html#the-if-else-class-of-statements-should-have-the-following-form

Reply:
Agreed and fixed.